library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java

```diff
@@ -141,6 +142,9 @@ private void init() {
         mPaint.setStrokeCap(Paint.Cap.ROUND);
 
         mInterpolator = new LinearInterpolator();
+
+        DisplayMetrics displayMetrics = getResources().getDisplayMetrics();
+        textViewVirtualWidth = displayMetrics.widthPixels * 2;
```

Comment:
Why was the ``* 2`` needed?

Reply:
I have updated this to use the texts width instead of the device's width this makes sense and works well.